testsuite/testsuite_support/python_driver.py

```diff
@@ -36,151 +24,34 @@ class PythonDriver(TestDriver):
-
-    def set_result_status(self, failed, message=''):
-        """
-        Shortcut to call `self.result.set_status`, taking the expected failure
-        message into account (if provided).
-
-        :param bool failed: Whether the testcase should be considered as
-            failed.
-        :param None|str message: Message for the status, if any.
-        """
-        if failed:
-            if self.expect_failure:
-                status = 'XFAIL'
-                message = '{} ({})'.format(message,
-                                           self.expect_failure_comment)
-            else:
-                status = 'FAILED'
-        else:
-            status = 'UOK' if self.expect_failure else 'PASSED'
-            message = self.expect_failure_comment
-
-        self.result.set_status(status, message)
-
     def run(self):
-        # Run the Python script and redirect its output to `self.out_file`.
-        argv = [self.python_interpreter] + self.python_interpreter_args
-        env = os.environ.copy()
-        env['EXT_SRC'] = self.external_sources_dir
+        env = dict(os.environ)
         env['target-triplet'] = str(self.env.target.triplet)
 
-        p = Run(argv, timeout=self.timeout, output=PIPE, error=STDOUT,
-                cwd=self.test_working_dir(), env=env)
-
-        with open(self.test_working_dir(self.out_file), 'a') as f:
-            f.write(p.out)
-
-        if p.status != 0:
-            self.result.actual_output += '{} returned status code {}\n'.format(
-                ' '.join(argv), p.status
-            )
-            self.result.actual_output += p.out
-            self.set_result_status(True, 'error status code')
-
-    def analyze(self):
-        diff = fileutils.diff(self.test_working_dir(self.expected_file),
-                              self.test_working_dir(self.out_file))
-
-        # Determine the status of this test, ignoring expected failure (for
-        # now).
-        if diff:
-            self.result.actual_output += diff
-            failed = True
-            message = 'diff in output'
-        else:
-            failed = False
-            message = ''
-
-        self.set_result_status(failed, message)
+        if 'target' in self.test_env and self.test_env['target'] != Env().target.triplet:
+            raise TestSkip('skip: test for %s' % self.test_env['target'])
+
+        # Also give it access to our Python helpers
+        python_path = env.get('PYTHONPATH', '')
+        path_for_drivers = os.path.abspath(
+            os.path.dirname(os.path.dirname(__file__)))
+        env['PYTHONPATH'] = '{}{}{}'.format(
+            path_for_drivers, os.path.pathsep, python_path
+        ) if python_path else path_for_drivers
+
+        p = self.shell([sys.executable, 'test.py'],
+                       env=env,
+                       cwd=self.test_env['working_dir'])
+
+        with open(os.path.join(self.test_env['working_dir'], self.out_file), 'a') as f:
```

Comment:
Code starting at this line seems to duplicate the behavior of `DiffTestDriver` (https://e3-testsuite.readthedocs.io/en/latest/api_diff.html): what about re-using that instead? This looks more future-proof, as you get also more features with that class.